src/policy/CMSRucioPolicy/permission.py

```diff
@@ -167,6 +174,128 @@ def perm_update_rse(issuer, kwargs, *, session: "Optional[Session]" = None):
     return _is_root(issuer) or has_account_attribute(account=issuer, key='admin', session=session)
 
 
+def _check_for_auto_approve_eligibility(issuer, rses, kwargs, session: "Optional[Session]" = None):
+
+    def _get_rule_size(rules):
+        rule_size = 0
+        for rule in rules:
+            scope = rule['scope']
+            name = rule['name']
+            rule_files = list_files(scope, name, session=session)
+            rule_size += sum([file['bytes'] for file in rule_files])
+        return rule_size
+
+    # prevent rule creation under 'User AutoApprove' for rules without ask_approval
+    if not kwargs["ask_approval"]:
+        return False
+    # prevent rule creation to tape and Tier3 and Tier0 under the 'User AutoApprove' activity
+    rule_rses = {rse['rse'] for rse in rses}
+    try:
+        t3_rses = {rse['rse'] for rse in parse_expression("tier=3|tier=0", session=session)}
+    except InvalidRSEExpression:
+        t3_rses = set()
+
+    try:
+        tape_rses = {rse['rse'] for rse in parse_expression("rse_type=TAPE", session=session)}
+    except InvalidRSEExpression:
+        tape_rses = set()
+
+    if rule_rses.intersection(t3_rses) or rule_rses.intersection(tape_rses):
+        return False
+
+    account = kwargs['account']
+    auto_approve_activity = 'User AutoApprove'
+    dids = kwargs['dids']
+    try:
+        global_usage_all_accounts = float(config_get(
+            'rules', 'global_usage_all_accounts', raise_exception=True, default=1e16))
+    except (NoOptionError, NoSectionError, RuntimeError):
+        global_usage_all_accounts = 1e16
+
+    try:
+        global_usage_per_account = float(config_get(
+            'rules', 'global_usage_per_account', raise_exception=True, default=1e15))
+    except (NoOptionError, NoSectionError, RuntimeError):
+        global_usage_per_account = 1e15
+
+    try:
+        rule_lifetime_threshold = int(config_get('rules', 'rule_lifetime_threshold',
+                                      raise_exception=True, default=2592000))
```

Comment:
Can you express this number as X*24*3600 so it's obvious how many days this is?

Reply:
Ok